pkg/health/listener.go

```diff
@@ -4,30 +4,43 @@ import (
 	"context"
 	"math"
 	"math/rand"
+	"os"
+	"strings"
 	"sync"
 	"time"
 
-	"golang.org/x/sync/semaphore"
-
 	"github.com/rancher/opni/pkg/agent"
+	"github.com/rancher/opni/pkg/alerting"
+	"github.com/rancher/opni/pkg/alerting/condition"
+	alertingv1alpha "github.com/rancher/opni/pkg/apis/alerting/v1alpha"
 	corev1 "github.com/rancher/opni/pkg/apis/core/v1"
 	"github.com/rancher/opni/pkg/auth/cluster"
 	"github.com/rancher/opni/pkg/util"
+	ap "github.com/rancher/opni/plugins/alerting/pkg/alerting"
+	"golang.org/x/sync/semaphore"
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/emptypb"
 )
 
 type Listener struct {
 	ListenerOptions
-	statusUpdate chan StatusUpdate
-	healthUpdate chan HealthUpdate
-	idLocksMu    sync.Mutex
-	idLocks      map[string]*sync.Mutex
-	closed       chan struct{}
-	sem          *semaphore.Weighted
+	statusUpdate        chan StatusUpdate
+	healthUpdate        chan HealthUpdate
+	idLocksMu           sync.Mutex
+	idLocks             map[string]*sync.Mutex
+	closed              chan struct{}
+	sem                 *semaphore.Weighted
+	AlertProvider       *alerting.Provider
```

Comment:
Pointer to interface is probably not what you want

Reply:
Here I think it might be, because this will be a pointer to the NOOP alerting.Provider interface that is passed down to async goroutines that need to keep a ref to the interface implementation when it gets updated elsewhere